hirundo/_http.py

```diff
@@ -0,0 +1,14 @@
+from requests import Response
+
+import hirundo.logger
+
+logger = hirundo.logger.get_logger(__name__)
+
+
+def raise_for_status_with_reason(response: Response):
+    try:
+        response.reason = response.json().get("reason", None)
+    except Exception as e:
+        logger.debug("Failed to parse response as JSON: %s", e)
```

Comment:
This won't be the output to the user, right? It's just a debug log

Reply:
If the user has the logger set to debug, it will show to the user. The default is `INFO`